client/my-sites/domains/domain-management/settings/cards/domain-security-details/index.tsx

```diff
@@ -1,15 +1,72 @@
+import { Button } from '@automattic/components';
 import { localizeUrl } from '@automattic/i18n-utils';
 import { CONTACT, HTTPS_SSL } from '@automattic/urls';
 import { Icon, lock } from '@wordpress/icons';
 import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import Accordion from 'calypso/components/domains/accordion';
+import useSslDetailsQuery from 'calypso/data/domains/ssl/use-ssl-details-query';
+import useProvisionCertificateMutation from 'calypso/data/domains/ssl/use-ssl-provision-certificate-mutation';
 import { sslStatuses } from 'calypso/lib/domains/constants';
+import { useDispatch } from 'calypso/state';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import { fetchSiteDomains } from 'calypso/state/sites/domains/actions';
 import { getSslReadableStatus, isSecuredWithUs } from '../../helpers';
-import type { DetailsCardProps } from '../types';
+import type { SecurityCardProps } from '../types';
 
 import './style.scss';
 
-const DomainSecurityDetails = ( { domain }: DetailsCardProps ) => {
+const noticeOptions = {
+	duration: 5000,
+	id: `ssl-status-notification`,
+};
+
+const DomainSecurityDetails = ( { domain, isDisabled, selectedSite }: SecurityCardProps ) => {
+	const dispatch = useDispatch();
+	// const calypsoDispatch = useCalypsoDispatch();
+
 	const translate = useTranslate();
+	const [ isExpanded, setIsExpanded ] = useState( false );
+
+	const {
+		data: sslDetails,
+		isFetching: isLoadingSSLData,
+		isError,
+		isStale,
+		refetch: refetchSSLStatusData,
+	} = useSslDetailsQuery( domain.name );
+
+	// Display success notices when SSL certificate is provisioned
+	const { provisionCertificate, isPending: isProvisioningCertificate } =
+		useProvisionCertificateMutation( domain.name, {
+			onSuccess() {
+				dispatch(
+					successNotice( translate( 'New certificate provisioning requested' ), noticeOptions )
+				);
+			},
+			onError( error ) {
+				dispatch( errorNotice( error.message, noticeOptions ) );
+			},
+		} );
+
+	// Render an error if ssl details fails to load
+	useEffect( () => {
+		if ( isError ) {
+			dispatch(
+				errorNotice(
+					translate( 'An error occurred while fetching ssl certificate details.' ),
+					noticeOptions
+				)
+			);
+		}
+	}, [ isError, dispatch, translate ] );
+
+	useEffect( () => {
+		if ( isStale && domain.sslStatus === sslStatuses.SSL_PENDING ) {
+			refetchSSLStatusData();
+			dispatch( fetchSiteDomains( selectedSite.ID ) );
+		}
+	}, [ isStale, domain.sslStatus, refetchSSLStatusData, selectedSite, dispatch ] );
 
 	if ( ! isSecuredWithUs( domain ) ) {
```

Comment:
This check will prevent the card from being shown if the domain is new and not yet resolving or if it's intended to point to a WPCOM side, but is misconfiugured.

I wonder if we should Just always show the card.

Reply:
You're a right! We discussed this, I forgot to remove the check. It's fixed now 👍